Woof/Woof/Model/Preferences/PreferencesHandler.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+
+/**
+ A type used for managing user preferences in an application.
+ */
+enum PreferencesHandler {
+    /**
+     Saves the selected role as a user preference.
+
+     - Parameters:
+        - selectedRole: The selected role to be saved.
+
+     - Returns: A boolean value indicating whether the saving operation was successful.
+     */
+    @discardableResult static func saveSelectedRole(_ selectedRole: Role) -> Bool {
+        var preferences: Preferences = loadPreferencesFromStorage() ?? Preferences()
+
+        preferences.selectedRole = selectedRole
+
+        return savePreferencesToStorage(preferences)
+    }
+
+    /**
+     Loads the selected role from user preferences.
+
+     - Returns: The selected role, or `.none` if no preferences are found.
+     */
+    static func loadSelectedRole() -> Role {
+        guard let preferences = loadPreferencesFromStorage() else { return .none }
+
+        let selectedRole = preferences.selectedRole
+
+        return selectedRole
```

Comment:
Did you mean:

```suggestion
        loadPreferencesFromStorage()?.selectedRole ?? .none
```

?

Reply:
Thanks for the suggestion, its looks more readable, I'll fix it.